src/plugins/data/common/query/text_based_query_state_to_ast.ts

```diff
@@ -57,18 +57,16 @@ export function textBasedQueryStateToExpressionAst({
 
   if (query && isOfAggregateQueryType(query)) {
     const mode = getAggregateQueryMode(query);
-    for (const esMode of ['sql', 'esql']) {
-      if (mode === esMode && esMode in query) {
-        const essql = aggregateQueryToAst({
-          query,
-          timeField: timeFieldName,
-          titleForInspector,
-          descriptionForInspector,
-        });
+    if (mode === 'esql') {
+      const essql = aggregateQueryToAst({
```

Comment:
I guess it is like that because this was the name of the expression. I changed it here https://github.com/elastic/kibana/pull/182079/commits/26b90692ad781191c0bd4b6c5589f82cffefe4ed